controllers/pkg/reconcilers/spire-bootstrap/reconciler.go

```diff
@@ -0,0 +1,475 @@
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, c any) (map[schema.GroupVersionKind]chan event.GenericEvent, error) {
+	r.Client = mgr.GetClient()
+
+	return nil, ctrl.NewControllerManagedBy(mgr).
+		Named("BootstrapSpireController").
+		For(&capiv1beta1.Cluster{}).
+		Complete(r)
+}
+
+type reconciler struct {
+	client.Client
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	cl := &capiv1beta1.Cluster{}
+	err := r.Get(ctx, req.NamespacedName, cl)
+	if err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			log.Error(err, "unable to fetch Cluster")
+		}
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Add your reconciliation logic here
+	log.Info("Reconciling Cluster", "cluster", cl.Name)
+
+	// Fetch the ConfigMap from the current cluster
+	configMapName := types.NamespacedName{Name: "spire-bundle", Namespace: "spire"}
+	configMap := &v1.ConfigMap{}
+	err = r.Get(ctx, configMapName, configMap)
+	if err != nil {
+		log.Error(err, "unable to fetch ConfigMap")
+		return reconcile.Result{}, err
+	}
+
+	secrets := &v1.SecretList{}
+	if err := r.List(ctx, secrets); err != nil {
+		msg := "cannot list secrets"
+		log.Error(err, msg)
+		return ctrl.Result{}, errors.Wrap(err, msg)
+	}
+
+	// Get the spire-server service
+	spireService := &v1.Service{}
+	err = r.Get(ctx, types.NamespacedName{Name: "spire-server", Namespace: "spire"}, spireService)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get spire-server service: %v", err)
+	}
+
+	// Get the ClusterIP
+	clusterIP := spireService.Spec.ClusterIP
+
+	// Get the port
+	var port string
+	if len(spireService.Spec.Ports) > 0 {
+		port = fmt.Sprint(spireService.Spec.Ports[0].Port)
+	}
+
+	// Construct the service address
+	serviceAddress := fmt.Sprintf("%s:%s", clusterIP, port)
+
+	spireAgentCM, err := createSpireAgentConfigMap("spire-agent", "spire", cl.Name, clusterIP, port)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to create spireAgent ConfigMap: %v", err)
+	}
+
+	for _, secret := range secrets.Items {
+		if strings.Contains(secret.GetName(), cl.Name) {
+			secret := secret // required to prevent gosec warning: G601 (CWE-118): Implicit memory aliasing in for loop
+			clusterClient, ok := cluster.Cluster{Client: r.Client}.GetClusterClient(&secret)
+			client, ready, err := clusterClient.GetClusterClient(ctx)
+			if err != nil {
+				msg := "cannot get clusterClient"
+				log.Error(err, msg)
+				return ctrl.Result{RequeueAfter: 30 * time.Second}, errors.Wrap(err, msg)
+			}
+			if ok {
+
+				kubeconfig := secret.Data["value"]
+				config, err := clientcmd.RESTConfigFromKubeConfig(kubeconfig)
+				if err != nil {
+					log.Error(err, "failed to get kubeconfig")
+				}
+				clientset, err := kubernetes.NewForConfig(config)
```

Comment:
So I'm generating a clientset here. I see nephio already has their own implementation of a K8s client. The reason for this I have commented on the createKubeConfig function